ocaml/testsuite/tests/typing-layouts/annots_beta.ml

```diff
@@ -0,0 +1,508 @@
+(* TEST
+   flags = "-extension layouts_beta"
+   * expect
+*)
+
+type t_value : value
+type t_imm : immediate
+type t_imm64 : immediate64
+type t_any : any;;
+
+[%%expect{|
+type t_value : value
+type t_imm : immediate
+type t_imm64 : immediate64
+Line 4, characters 13-16:
+4 | type t_any : any;;
+                 ^^^
+Error: Layout any is used here, but the appropriate layouts extension is not enabled
+|}]
+(* CR layouts v1.5: the above test should be accepted; fix. *)
+
+type t_void : void;;
+
+[%%expect{|
+Line 1, characters 14-18:
+1 | type t_void : void;;
+                  ^^^^
+Error: Layout void is used here, but the appropriate layouts extension is not enabled
+|}]
+
+(***************************************)
+(* Test 1: annotation on type variable *)
+
+let x : int as ('a: value) = 5
+let x : int as ('a : immediate) = 5
+let x : int as ('a : any) = 5;;
```

Comment:
Yes, and that's why there's a CR layouts v1.5 saying that it should be accepted above.

The problem is that `type t_any : any` is parsed into `type t_any [@@any]`, which uses a different mechanism than the one all the other layouts flow through here. There is a ticket in our internal system to move that over to the uniform mechanism, at which point this will get fixed. It didn't seem necessary to rip out the existing mostly-working functionality in the same PR that all this new stuff was added.

Given the existence of the CR saying this needs to be fixed soon, I'm resolving this Conversation, but please reopen if you disagree.